### PR TITLE
889191: Drafted a content for csp.

### PR DESCRIPTION
**Pull request by ArunKumar-SF3979**
# Description

- Need to update the csp content and move it to separate page.

### CODE REVIEW

**Comment by ArunKumar-SF3979** [comment]
rebuild

File: ej2-asp-core-mvc/common/EJ2_ASP.NETCORE/content-security-policy.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+Content Security Policy (CSP) is a security feature implemented by web browsers to protect against attacks such as cross-site scripting (XSS) and data injection. It limits the sources from which content can be loaded on a web page. To enable strict Content Security Policy (CSP), certain browser features are disabled by default. To use Syncfusion controls with strict CSP mode, it is essential to include the following directives:
+
+Nonce attribute is used in CSP to validate requests and prevent attackers from injecting JavaScript code into a web application.
+
+
+## Steps to Create and Include the Nonce Attribute in an ASP.NET Core Application
+
+* Generate the `nonce` attribute value by adding the below code in **Program.cs** file.
+
+* Syncfusion material and tailwind built-in themes contain a reference to the Roboto’s external font, which is also blocked. To allow them, add the external font reference to the style-src and font-src directives in the above meta tag.
```
Comment (kmuthukumarmkm) [inline ej2-asp-core-mvc/common/EJ2_ASP.NETCORE/content-security-policy.md:22]: high light it.

`style-src` and `font-src`

File: ej2-asp-core-mvc/common/EJ2_ASP.NETCORE/content-security-policy.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+control: Common
+publishingplatform: ##Platform_Name##
+documentation: ug
+---
+
+# Content Security Policy
+
+Content Security Policy (CSP) is a security feature implemented by web browsers to protect against attacks such as cross-site scripting (XSS) and data injection. It limits the sources from which content can be loaded on a web page. To enable strict Content Security Policy (CSP), certain browser features are disabled by default. To use Syncfusion controls with strict CSP mode, it is essential to include the following directives:
+
+Nonce attribute is used in CSP to validate requests and prevent attackers from injecting JavaScript code into a web application.
```
Comment (kmuthukumarmkm) [inline ej2-asp-core-mvc/common/EJ2_ASP.NETCORE/content-security-policy.md:15]: `nonce`

File: ej2-asp-core-mvc/common/EJ2_ASP.NETCORE/content-security-policy.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+{% highlight c# tabtitle="~/_Layout.cshtml" %}
+<body>
+    ...
+    <!-- Syncfusion ASP.NET Core Script Manager -->
+    <ejs-scripts add-nonce="@Context.Items["ScriptNonce"]"></ejs-scripts>
+</body>
+{% endhighlight %}
+{% endtabs %}
+
+N> The `unsafe-eval` directive needs to be added in script-src for samples that use the template properties.
```
Comment (kmuthukumarmkm) [inline ej2-asp-core-mvc/common/EJ2_ASP.NETCORE/content-security-policy.md:73]: `script-src`

samples that use the template properties -> this term seems un relatable. 

controls which uses the templates

File: ej2-asp-core-mvc/common/EJ2_ASP.NETCORE/content-security-policy.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
+---
+layout: post
+title: Content Security Policy in ##Platform_Name## Common Control | Syncfusion
+description: Learn here all about Content Security Policy in Syncfusion ##Platform_Name## Common control of Syncfusion Essential JS 2 and more.
```
Comment (kmuthukumarmkm) [inline ej2-asp-core-mvc/common/EJ2_ASP.NETCORE/content-security-policy.md:4]: Syncfusion Essential JS 2 -> is this right? or need to correct?

File: ej2-asp-core-toc.html [hunk 5]
```diff
@@ -107,6 +107,7 @@
 			</li>
 			<li><a href="/ej2-asp-core/common/drag-and-drop">Drag and Drop</a></li>
 			<li><a href="/ej2-asp-core/common/compatible-with-essential-JS1">EJ1 and EJ2 Compatibility</a></li>
+			<li><a href="/ej2-asp-core/common/content-security-policy">Content Security Policy</a></li>
```
Comment (kmuthukumarmkm) [inline ej2-asp-core-toc.html:110]: what about mvc-toc? 

Did we checked with MVC controls?

All this content prepared based on local try out. right?

File: ej2-asp-core-mvc/common/EJ2_ASP.NETCORE/content-security-policy.md (hunk 1 above)
Comment (ArunKumar-SF3979) [reply ej2-asp-core-mvc/common/EJ2_ASP.NETCORE/content-security-policy.md]: Highlighted.

File: ej2-asp-core-mvc/common/EJ2_ASP.NETCORE/content-security-policy.md (hunk 4 above)
Comment (ArunKumar-SF3979) [reply ej2-asp-core-mvc/common/EJ2_ASP.NETCORE/content-security-policy.md:4]: all md files uses the same

**Comment by ArunKumar-SF3979** [comment]
rebuild

**Comment by ArunKumar-SF3979** [comment]
rebuild

**Review by kmuthukumarmkm — APPROVED** [review]

**Pull request merged**